Implement url / word whitelist

Innectic opened the pull request:
I wonder if para will even accept this :thinking:

Innectic committed:
Implement url / word whitelist

Innectic committed:
Add blacklist

Innectic committed:
Oops

Innectic committed:
Add blacklist to the schema

RPiAwesomeness commented on app/schemas/config.py:
```diff
@@ -31,10 +31,21 @@ class SpamSchema(Schema):
 
 
+class WhitelistSchema(Schema):
+    whitelisted_urls = fields.List(fields.String())
```
`whitelisted_urls = fields.List(fields.String(), default=[])`

Same for `_words` and backlists

RPiAwesomeness commented on app/models/config.py:
```diff
@@ -23,7 +23,7 @@ def default_data(token=None):
             "announce": Announcements().__dict__,
             "spam": Spam().__dict__,
-            "whitelisted_urls": []
+            "whitelist": Whitelist().__dict__
```
`"whitelist": [Whitelist().__dict__]` since it's a dictionary. Same for blacklist.

RPiAwesomeness committed:
Merge branch 'develop' into feature/whitelist

RPiAwesomeness committed:
Fix the things because Innectic is lazy :mappa

RPiAwesomeness committed:
Whitelist/Blacklist passing UT!
Changed UT to delete all non-root users and keys on failure instead of
leaving them as it did previously

RPiAwesomeness commented:
We'll leave this on that separate branch until we can do some staged testing.
